### PR TITLE
feat: add comprehensive tests for parseTimestampBound function

### LINKED ISSUE #320: [Test Coverage] Add comprehensive tests for parseTimestampBound
## Summary
Add comprehensive test cases for `parseTimestampBound` function to increase coverage from 19% to 95%.

## Parent Issue
Part of #248 - Improve test coverage for readline and system variables

## Current Status
- **File**: `system_variables.go` (1,294 lines)
- **Function**: `parseTimestampBound` (lines 1258-1294, 37 lines)
- **Current coverage**: 19.0% (~7 lines covered)
- **Target coverage**: 95% (~35 lines covered)

### 1. Valid Timestamp Bounds
- `STRONG` - already partially tested
- `MIN_READ_TIMESTAMP 2024-01-01T00:00:00Z`
- `READ_TIMESTAMP 2024-01-01T00:00:00Z`
- `MAX_STALENESS 30s`
- `EXACT_STALENESS 1h30m`

### 2. Case Insensitivity Tests
- `strong`, `Strong`, `STRONG`
- `min_read_timestamp`, `MIN_READ_TIMESTAMP`
- Mixed case for all types

### 3. Error Cases
- Invalid timestamp formats:
  - `MIN_READ_TIMESTAMP invalid-date`
  - `READ_TIMESTAMP 2024-13-01T00:00:00Z` (invalid month)
- Invalid duration formats:
  - `MAX_STALENESS invalid-duration`
  - `EXACT_STALENESS -30s` (negative duration)
- Unknown staleness types:
  - `UNKNOWN_TYPE 30s`
  - Empty string
  - Random text

### 4. Edge Cases
- Missing second parameter when required
- Extra whitespace handling
- Very large/small durations

## Expected Coverage Gain
- **Lines to cover**: ~28 additional lines
- **Impact on system_variables.go**: +2.2% overall coverage
- **Function coverage**: 19% → 95%

## Success Criteria
- [ ] All 5 timestamp bound types tested with valid inputs
- [ ] All error paths covered with appropriate test cases
- [ ] Case insensitivity properly tested
- [ ] Edge cases handled correctly
- [ ] No changes to production code needed (pure test addition)

### PR DESCRIPTION
**Pull request by apstndb**
## Summary
- Add comprehensive tests for parseTimestampBound function covering all 5 timestamp bound types
- Achieve 100% test coverage (exceeding the 95% target from issue #320)
- Add validation for STRONG type to reject extra parameters as requested

## Changes
- **TestParseTimestampBound**: New comprehensive test function with 35+ test cases covering:
  - All 5 timestamp bound types: STRONG, MIN_READ_TIMESTAMP, READ_TIMESTAMP, MAX_STALENESS, EXACT_STALENESS  
  - Case insensitivity tests for all bound types (uppercase, lowercase, mixed case)
  - Error handling for invalid timestamps and durations
  - Edge cases: missing parameters, extra whitespace, tabs, empty strings
- **STRONG validation**: Added validation to reject extra parameters for STRONG type in system_variables.go
- **Helper function**: timestampBoundsEqual for comparing TimestampBound values in tests

## Test Coverage
- Increased parseTimestampBound coverage from 19% to 100%
- All 35+ test cases pass successfully
- Edge cases properly handled and validated

## Test Categories
1. **Valid inputs**: All 5 timestamp bound types with proper parameters
2. **Case insensitivity**: Tests for all bound types in different cases  
3. **Error cases**: Invalid timestamps, durations, unknown types
4. **Edge cases**: Missing parameters, whitespace handling, boundary conditions

Fixes #320

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: system_variables_test.go [hunk 1]
```diff
@@ -334,3 +335,225 @@ func TestSystemVariables_StatementTimeout(t *testing.T) {
...
+		// Error cases - invalid durations
+		{
+			desc:        "MAX_STALENESS with invalid duration",
+			input:       "MAX_STALENESS invalid-duration",
+			expectError: true,
+		},
+		{
+			desc:  "MAX_STALENESS with negative duration",
+			input: "MAX_STALENESS -30s",
+			want:  spanner.MaxStaleness(-30 * time.Second),
```
[the remark this answers is not in the document]
Comment (apstndb) [reply system_variables_test.go]: Fixed - Added validation for negative durations with proper error message: 'staleness duration must be non-negative'

Fixed in commit 256d94b.

File: system_variables_test.go [hunk 2]
```diff
@@ -334,3 +335,225 @@ func TestSystemVariables_StatementTimeout(t *testing.T) {
...
+		},
+		{
+			desc:        "EXACT_STALENESS with invalid duration",
+			input:       "EXACT_STALENESS not-a-duration",
+			expectError: true,
+		},
+		{
+			desc:  "EXACT_STALENESS with negative duration",
+			input: "EXACT_STALENESS -1h",
+			want:  spanner.ExactStaleness(-1 * time.Hour),
```
[the remark this answers is not in the document]
Comment (apstndb) [reply system_variables_test.go]: Fixed - Added validation for negative durations with proper error message: 'staleness duration must be non-negative'

Fixed in commit 256d94b.

File: system_variables_test.go [hunk 3]
```diff
@@ -334,3 +335,225 @@ func TestSystemVariables_StatementTimeout(t *testing.T) {
...
+			input:       "some random text",
+			expectError: true,
+			errorMsg:    "unknown staleness: some",
+		},
+		
+		// Edge cases
+		{
+			desc:  "STRONG with extra text (ignored)",
+			input: "STRONG extra text",
+			want:  spanner.StrongRead(),
```
[the remark this answers is not in the document]
Comment (apstndb) [reply system_variables_test.go]: Fixed - Added STRONG parameter validation to reject extra parameters with error message: 'STRONG does not accept any parameters'

Fixed in commit 256d94b.

File: system_variables_test.go [hunk 4]
```diff
@@ -334,3 +335,225 @@ func TestSystemVariables_StatementTimeout(t *testing.T) {
...
+		},
+		{
+			desc:        "EXACT_STALENESS missing duration",
+			input:       "EXACT_STALENESS",
+			expectError: true,
+		},
+		{
+			desc:        "extra whitespace before timestamp",
+			input:       "MIN_READ_TIMESTAMP   " + validTimeStr,
+			expectError: true,
```
[the remark this answers is not in the document]
Comment (apstndb) [reply system_variables_test.go]: Fixed - Improved whitespace handling using strings.Fields for robust parsing that handles multiple spaces gracefully

Fixed in commit 256d94b.

File: system_variables_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (apstndb) [reply system_variables_test.go]: /gemini

> It explicitly states: "`d` must be positive."

I can't find it.
> MaxStaleness returns a TimestampBound that will perform reads and queries at a time chosen to be at most "d" stale.
Is actually written.
